ada_2_fhir-r4/zibs2020/payload/0.0/nl-core-Alert.xsl

```diff
@@ -0,0 +1,181 @@
+                                <xsl:attribute name="value">
+                                    <xsl:call-template name="format2FHIRDate">
+                                        <xsl:with-param name="dateTime" select="xs:string(begin_datum_tijd/@value)"/>
+                                    </xsl:call-template>
+                                </xsl:attribute>
+                            </start>
+                        </xsl:if>
+                        <xsl:if test="eind_datum_tijd">
+                            <end>
+                                <xsl:attribute name="value">
+                                    <xsl:call-template name="format2FHIRDate">
+                                        <xsl:with-param name="dateTime" select="xs:string(eind_datum_tijd/@value)"/>
+                                    </xsl:call-template>
+                                </xsl:attribute>
+                            </end>
+                        </xsl:if>
+                    </period>
+                </xsl:if>
+            </Flag>
+        </xsl:for-each>
+    </xsl:template>
+    
+    <xd:doc>
+        <xd:desc>Template to generate a unique id to identify this instance.</xd:desc>
+    </xd:doc>
+    <xsl:template match="alert" mode="_generateId">
+        <xsl:variable name="parts">
+            <xsl:text>Alert</xsl:text>
+            <xsl:value-of select="alert_type/@displayName"/>
+            <xsl:value-of select="alert_naam/@displayName"/>
+            <xsl:if test="begin_datum_tijd/@value">
+                <xsl:value-of select="concat('start datum', begin_datum_tijd/@value)"/>
+            </xsl:if>
+            <xsl:if test="eind_datum_tijd/@value">
+                <xsl:value-of select="concat('eind datum', eind_datum_tijd/@value)"/>
+            </xsl:if>
+            <xsl:value-of select="toelichting/@value"/>
+        </xsl:variable>
+        <xsl:value-of select="substring(replace(string-join($parts, '-'), '[^A-Za-z0-9-.]', ''), 1, 64)"/>
+    </xsl:template>
+    
+    <xd:doc>
+        <xd:desc>Template to generate a display that can be shown when referencing this instance.</xd:desc>
+    </xd:doc>
+    <xsl:template match="alert" mode="_generateDisplay">
+        <xsl:variable name="parts">
+            <xsl:text>Alert</xsl:text>
+            <xsl:value-of select="alert_type/@displayName"/>
+            <xsl:value-of select="alert_naam/@displayName"/>
+            <xsl:value-of select="concat(waarneming_gebruik/hoeveelheid/@value, waarneming_gebruik/hoeveelheid/@unit)"/>
+            <xsl:if test="begin_datum_tijd/@value">
+                <xsl:value-of select="concat('start datum', begin_datum_tijd/@value)"/>
+            </xsl:if>
+            <xsl:if test="eind_datum_tijd/@value">
+                <xsl:value-of select="concat('eind datum', eind_datum_tijd/@value)"/>
+            </xsl:if>
+        </xsl:variable>
+        <xsl:value-of select="string-join($parts, ', ')"/>     
+    </xsl:template>
```

Comment:
Zie opmerkingen bij Tobacco

Reply:
Aangepast, hier zaten ook nog kopierfoutjes in!